internal/store/atlas/api_keys.go

```diff
@@ -37,75 +39,92 @@ type ProjectAPIKeyAssigner interface {
 }
 
 type OrganizationAPIKeyLister interface {
-	OrganizationAPIKeys(string, *atlas.ListOptions) ([]atlas.APIKey, error)
+	OrganizationAPIKeys(string, *atlas.ListOptions) (*atlasv2.PaginatedApiApiUser, error)
 }
 
 type OrganizationAPIKeyDescriber interface {
-	OrganizationAPIKey(string, string) (*atlas.APIKey, error)
+	OrganizationAPIKey(string, string) (*atlasv2.ApiUser, error)
 }
 
 type OrganizationAPIKeyUpdater interface {
-	UpdateOrganizationAPIKey(string, string, *atlas.APIKeyInput) (*atlas.APIKey, error)
+	UpdateOrganizationAPIKey(string, string, *atlas.APIKeyInput) (*atlasv2.ApiUser, error)
 }
 
 type OrganizationAPIKeyCreator interface {
-	CreateOrganizationAPIKey(string, *atlas.APIKeyInput) (*atlas.APIKey, error)
+	CreateOrganizationAPIKey(string, *atlas.APIKeyInput) (*atlasv2.ApiUser, error)
 }
 
 type OrganizationAPIKeyDeleter interface {
 	DeleteOrganizationAPIKey(string, string) error
 }
 
 // OrganizationAPIKeys encapsulates the logic to manage different cloud providers.
-func (s *Store) OrganizationAPIKeys(orgID string, opts *atlas.ListOptions) ([]atlas.APIKey, error) {
-	result, _, err := s.client.APIKeys.List(s.ctx, orgID, opts)
+func (s *Store) OrganizationAPIKeys(orgID string, opts *atlas.ListOptions) (*atlasv2.PaginatedApiApiUser, error) {
+	result, _, err := s.clientv2.ProgrammaticAPIKeysApi.ListApiKeys(s.ctx, orgID).
+		ItemsPerPage(int32(opts.ItemsPerPage)).PageNum(int32(opts.PageNum)).IncludeCount(opts.IncludeCount).Execute()
```

Comment:
can you please update the flags type instead of doing casting here?

Reply:
I've done this in a few other places for merged PRs so will amend them all in a separate PR after this one is merged https://github.com/mongodb/go-client-mongodb-atlas/pull/464